posthog/warehouse/data_load/validate_schema.py

```diff
@@ -109,8 +154,27 @@ async def validate_schema_and_update_table(run_id: str, team_id: int, schemas: l
         if not table_created:
             table_created = await acreate_datawarehousetable(external_data_source_id=job.pipeline.id, **data)
 
-        # TODO: this should be async too
-        table_created.columns = await sync_to_async(table_created.get_columns)()
+        for schema in table_schema.values():
```

Comment:
NIT: Kind of confused of with the structure of the logic here. We're iterating for a match and breaking after that match is handled? I think this can just be a dict lookup?

Reply:
Yeah, the key of `table_schemas` is not the same as the `resource` sadly. I could convert it to a dict, but I'm not so sure if we'd get much from that. I could use `filter(...)`, but I think this logic is easier to reason with